driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java

```diff
@@ -594,10 +594,10 @@ public boolean onTimeout(Connection connection, long latency, int retryCount) {
         }
 
         Host queriedHost = current;
+        // If a query times out, we consider that the host is unstable, so we defunct
+        // the connection to mark it down.
+        OperationTimedOutException timeoutException = new OperationTimedOutException(connection.address);
```

Comment:
True, I added it for JAVA-651 where it was only used for internal statements. But I agree that it's a good idea to use it here as well.